srtcore/core.cpp

```diff
@@ -8059,21 +8059,27 @@ int CUDT::sendCtrlAck(CPacket& ctrlpkt, int size)
     string reason; // just for "a reason" of giving particular % for ACK
 #endif
 
-    // If there is no loss, the ACK is the current largest sequence number plus 1;
-    // Otherwise it is the smallest sequence number in the receiver loss list.
-    if (m_pRcvLossList->getLossLength() == 0)
-    {
-        ack = CSeqNo::incseq(m_iRcvCurrSeqNo);
-#if ENABLE_HEAVY_LOGGING
-        reason = "expected next";
-#endif
-    }
-    else
     {
+        // If there is no loss, the ACK is the current largest sequence number plus 1;
+        // Otherwise it is the smallest sequence number in the receiver loss list.
         ScopedLock lock(m_RcvLossLock);
         ack = m_pRcvLossList->getFirstLostSeq();
+
+        // We don't need to check the length prematurely,
```

Comment:
Consider reducing the scope of lock to the target function `getFirstLostSeq()` only, thus also removing extra nested code block, although gaining one more operation line.

```suggestion
        // If there is no loss, the ACK is the current largest sequence number plus 1;
        // Otherwise it is the smallest sequence number in the receiver loss list.
        enterCS(m_RcvLossLock);
        ack = m_pRcvLossList->getFirstLostSeq();
        leaveCS(m_RcvLossLock);

        // We don't need to check the length prematurely,
```